pangeo_forge_recipes/transforms.py

```diff
@@ -429,8 +429,8 @@ class CombineReferences(beam.PTransform):
       dimensionality will match that of the accumulator.
     """
 
-    concat_dims: List[str]
-    identical_dims: List[str]
+    concat_dims: List[str] = field(default_factory=list)
+    identical_dims: List[str] = field(default_factory=list)
```

Comment:
Let's not use default factories here if at all possible.

IIUC these are required, so if they are not passed, we _do want_ a TypeError to be raised (which the default factory will prevent).

See related comment below.